src/linker/Linker.Dataflow/JsonFlowAnnotationSource.cs

```diff
@@ -0,0 +1,204 @@
+
+						var type = assembly.MainModule.GetType (namespaceName, typeName);
+						if (type == null) {
+							context.LogMessage ($"Type {namespaceName}.{typeName} couldn't be resolved");
+							continue;
+						}
+
+						foreach (var member in typeElement.Value.EnumerateObject ()) {
+							string memberName = member.Name;
+
+							if (member.Value.ValueKind == JsonValueKind.Object) {
+								// This is a method
+
+								// Parser uses same format as PreserveDependencyAttribute
+								string [] signature = null;
+								memberName = memberName.Replace (" ", "");
+								var sign_start = memberName.IndexOf ('(');
+								var sign_end = memberName.LastIndexOf (')');
+								if (sign_start > 0 && sign_end > sign_start) {
+									var parameters = memberName.Substring (sign_start + 1, sign_end - sign_start - 1);
+									signature = string.IsNullOrEmpty (parameters) ? Array.Empty<string> () : parameters.Split (',');
+									memberName = memberName.Substring (0, sign_start);
+								}
+
+								MethodDefinition method = null;
+								foreach (var candidate in type.Methods) {
+									if (candidate.Name != memberName)
+										continue;
+
+									if (signature != null) {
+										if (candidate.Parameters.Count != signature.Length)
+											continue;
+
+										bool sigMatch = true;
+										for (int i = 0; i < candidate.Parameters.Count; i++) {
+											if (candidate.Parameters [i].ParameterType.FullName != signature [i].ToCecilName ()) {
+												sigMatch = false;
+												break;
+											}
+										}
+
+										if (!sigMatch)
+											continue;
+									}
+
+									if (method != null) {
+										context.LogMessage ($"Multiple matches for method {memberName}");
+									}
+
+									method = candidate;
+								}
+
+								if (method == null) {
+									context.LogMessage ($"No match for {memberName}");
+									continue;
+								}
+
+								DynamicallyAccessedMemberKinds returnAnnotation = 0;
+								var parameterAnnotations = new ArrayBuilder<(string ParamName, DynamicallyAccessedMemberKinds Annotation)> ();
+								foreach (var parameter in member.Value.EnumerateObject ()) {
+									if (parameter.Name == "return")
+										returnAnnotation = ParseKinds (parameter.Value.GetString ());
+									else
+										parameterAnnotations.Add ((parameter.Name, ParseKinds (parameter.Value.GetString ())));
+								}
+
+								_methods [method] = new AnnotatedMethod (returnAnnotation, parameterAnnotations.ToArray ());
+							} else {
+								// This is a field or property
+								FieldDefinition field = null;
+								foreach (var candidate in type.Fields) {
+									if (candidate.Name != memberName)
+										continue;
+
+									// IL allows overloaded fields, but not worth adding messages for that...
+									field = candidate;
+									break;
+								}
+
+								if (field != null) {
+									_fields [field] = ParseKinds (member.Value.GetString ());
+								}
```

Comment:
we can `break` here right? (no property should match below)

Reply:
I think `continue` works better, but it's very far away from where the loop is and it felt a bit dangerous to add it. Nevertheless, I added it.